starter-code/public/scripts/article.js

```diff
@@ -3,128 +3,161 @@ var app = app || {};
+    the function.
+    As a result, we no longer have to pass in the optional "this" argument to forEach!*/
+    Object.keys(rawDataObj).forEach(key => this[key] = rawDataObj[key]);
+  }
+
+  Article.all = [];
+
+  Article.prototype.toHtml = function() {
+    var template = Handlebars.compile($('#article-template').text());
+
+    this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+    this.publishStatus = this.publishedOn ? `published ${this.daysAgo} days ago` : '(draft)';
+    this.body = marked(this.body);
+
+    return template(this);
+  };
+
+  Article.loadAll = rows => {
+    rows.sort((a,b) => (new Date(b.publishedOn)) - (new Date(a.publishedOn)));
+
+    // DONE: Refactor this forEach code, by using a `.map` call instead, since what we are trying to accomplish
+    // is the transformation of one collection into another. Remember that we can set variables equal to the result
+    // of functions. So if we set a variable equal to the result of a .map, it will be our transformed array.
+    // There is no need to push to anything.
+
+    /* OLD forEach():
+    rawData.forEach(function(ele) {
+    Article.all.push(new Article(ele));
+  });
+  */
+
+    Article.all = rows.map(function(ele) {
+      return new Article(ele)
+    })
+  };
+
+  Article.fetchAll = callback => {
+    $.get('/articles')
+    .then(
+      results => {
+        Article.loadAll(results);
+        callback();
+      }
+    )
+  };
+
+  // DONE: Chain together a `map` and a `reduce` call to get a rough count of all words in all articles.
+  Article.numWordsAll = () => {
+    return Article.all.map(function(article){
+      return article.body.split(' ').length;
+    }).reduce(function(sum, value) {
+      return sum + value;
+    }, 0)
+  };
+
+  // DONE: Chain together a `map` and a `reduce` call to produce an array of unique author names. You will
+  // probably need to use the optional accumulator argument in your reduce call.
+  Article.allAuthors = () => {
+    return Article.all.map(function(article){
+      return article.author;
+    }).reduce(function(accumulator, author){
+      if (!accumulator.includes(author)){
+        accumulator.push(author)
+        return accumulator
+      } else {
+        return accumulator;
+      }
+    },[]);
+  };
+
+  Article.numWordsByAuthor = () => {
+    return Article.allAuthors().map(author => {
+      // DONE: Transform each author string into an object with properties for
+      // the author's name, as well as the total number of words across all articles
+      // written by the specified author.
+      // HINT: This .map should be setup to return an object literal with two properties.
+      // The first property should be pretty straightforward, but you will need to chain
+      // some combination of filter, map, and reduce to get the value for the second
+      // property.
+      return {
+        name: author,
+        totalWords: Article.all.filter(function(article) {
```

Comment:
This is not working 100%. If I open up the admin page, it gives me low numbers, less than a thousand words per author. But your Total words is 165311. 
Let me know if you need help with this